sketch.js

```diff
@@ -0,0 +1,73 @@
+class ColorHelper {
+  generateRandomColor() {
+    return [random(255), random(255), random(255)];
+  }
+}
+
+class Circle {
+  constructor(x, y, size, colorHelper) {
+    this.x = x;
+    this.y = y;
+    this.size = size;
+    this.speedX = random(1, 5);
+    this.speedY = random(1, 5);
+    this.colorHelper = colorHelper;
+    this.color = this.colorHelper.generateRandomColor();
+  }
+
+  setSpeed(speedX, speedY) {
+    this.speedX = speedX;
+    this.speedY = speedY;
+  }
+
+  moveCircle() {
+    this.x += this.speedX;
+    this.y += this.speedY;
+
+    if (this.x > width - this.size / 2 || this.x < this.size / 2) {
```

Comment:
You have a bug with this. If a circle was created overlapping with the edge, it stays there and only moves along one axis.